### ISSUE TITLE
Inconsistent default behavior for Jpegli

### ISSUE THREAD

**Issue by Galaxy4594** [libjxl/libjxl#3605]
The default for Jpegli is to output YUV as 4:4:4 regardless of quality level. However, according to exiftool, XYB outputs as 4:2:0 (blue-channel subsampled) as the default setting.

This can be confusing because, based on the YUV mode, one might assume that XYB also has full chroma resolution. When I noticed a slight color shift while compressing in XYB, I initially thought it was an encoder quirk. However, it turns out that XYB is indeed chroma subsampled.

For consistency, Jpegli should have both modes default to 4:4:4 chroma. A secondary benefit of this change would be that XYB jpegs can be transcoded to Jpegxl.

Another issue is present with XYB: the colors appear significantly different between the default 4:2:0 and the manually defined `chroma_subsampling=420`. My guess is that it’s using YUV subsampling rather than the XYB blue-channel subsampling.
The behavior is pretty strange and it doesn't really make much sense.

![animated](https://github.com/libjxl/libjxl/assets/164440799/56b6cd35-0473-42f0-9feb-bd1df141a6b2)

**Comment by j7nj7n** [comment]
The default mode subsamples component 3 "B" by 2x2, while --chroma_subsampling=420 subsamples both components 2 "G" and 3 "B" resulting into a greater error. The white glow is barely visible, and there is no obvious smearing between red and blue or erasure of fine red details even on difficult images. I think they wanted to boast a high compression ratio with this mode. You can make a batch script with 444.

XYB encoded example, needs ICC, you can barely see a glow around the letters:
![512_16_xyb_ss_LI](https://github.com/libjxl/libjxl/assets/31337218/78810083-fd60-4f76-af53-d6c4fba848f6)

**Comment by goodusername123** [comment]
A semi big problem with doing sub-sampling of any kind with XYB JPEGs is that the result is unable to be transcoded to JXL as the JPEG XL spec does not support RGB JPEGs with chroma subsampling.
It seems like this is the return of issue https://github.com/libjxl/libjxl/issues/2284 which was thought to have been solved.

**Comment by gcp** [comment]
>The default for Jpegli is to output YUV as 4:4:4 regardless of quality level.

FWIW it also depends on whether you use the command line tool or the library. The command line tool uses 4:4:4, the library defaults to 4:2:0.

**Comment by andrews05** [comment]
> Another issue is present with XYB: the colors appear significantly different between the default 4:2:0 and the manually defined `chroma_subsampling=420`. My guess is that it’s using YUV subsampling rather than the XYB blue-channel subsampling.
> The behavior is pretty strange and it doesn't really make much sense.

I've noticed this myself and you're right, it doesn't make sense, especially as you can't reproduce the "default" behaviour with any of the allowed values for `chroma_subsampling`.
I think the sensible behaviour for XYB would be to make the `chroma_subsampling` setting only apply to the B channel and not the Y/G channel.

**Comment by Jaldebehot** [comment]
I disagree. Jpegli should not be creating xyb/rgb jpegs with chroma subsampling, even if the user asks for it. It's not normal or expected for an RGB jpeg to have subsampling. Even libjxl does not expect or support it for lossless transcode! RGB JPEG is an Adobe extension to the JPEG standard and requires an app14 marker to be present in the file, which is the subject of a separate bug report involving the xyb mode.
